crud/src/main/java/com/redhat/lightblue/rest/crud/cmd/AbstractLockCommand.java

```diff
@@ -98,13 +103,19 @@ public CallStatus run() {
             o.set("result", result);
             return new CallStatus(new SimpleJsonObject(o));
         } catch (Error e) {
+            metricCtx.markRequestException(e, e.getErrorCode());
             LOGGER.error("failure: {}", e);
             return new CallStatus(e);
         } catch (Exception e) {
+            metricCtx.markRequestException(e, e.getMessage());
             LOGGER.error("failure: {}", e);
             return new CallStatus(Error.get(RestCrudConstants.ERR_REST_ERROR, e.toString()));
```

Comment:
So I think this case will still benefit from using the Error instance so it's consistent with the response like other Errors that we handle, like so:

```
              LOGGER.error("failure: {}", e);
              Error error = Error.get(RestCrudConstants.ERR_REST_ERROR, e.toString());
              metricCtx.markRequestException(error);
              return new CallStatus(error);
```

This implies an overload like:

```
        @Override
        public void markRequestException(Error e) {
            markRequestException(e, e.getErrorCode());
        }
```

Which I think is a good idea anyway. We want to encapsulate as much as possible, not rely on clients to know how to map an Error to marking exceptions.

Would use that for above catch too.

Reply:
I misunderstood your earlier comment regarding the same.Thanks for clarification and example. Update it as suggested.